src/mscorlib/src/System/Threading/ThreadPool.cs

```diff
@@ -1387,7 +1387,7 @@ private static object[] ToObjectArray(IEnumerable<IThreadPoolWorkItem> workitems
             i = 0;
             foreach (IThreadPoolWorkItem item in workitems)
             {
-                if (i < result.Length) //just in case someone calls us while the queues are in motion
+                if ((uint)i < (uint)result.Length) //just in case someone calls us while the queues are in motion
```

Comment:
Eh, we do interface calls but we're trying to get rid of a range check, I doubt that this will matter. Granted, it doesn't hurt either, at least the code will be smaller.